app/models/user.rb

```diff
@@ -81,6 +81,19 @@ def self.token_authentication_key
     :auth_token
   end
 
+  def reset_password(new_password, new_password_confirmation)
```

Comment:
I haven't tested this much, but how about something like:
```ruby
self.class.validators_on(:password).map { |v| v.validate_each(self, :password, self.password) }
return false if user.errors.any?
save(validate: false)
```

It doesn't seem like there are any validators installed on password_confirmation, but maybe there's some magic that I'm not aware of.